tests/lib/rules/no-restricted-imports.js

```diff
@@ -1953,6 +2008,119 @@ ruleTester.run("no-restricted-imports", rule, {
             endColumn: 9,
             message: "* import is invalid because import name matching '/^Foo/u' pattern from 'foo' is restricted from being used."
         }]
+    },
+    {
+        code: "export { Bar } from 'foo';",
+        options: [{
+            patterns: [{
+                group: ["foo"],
+                allowImportNamePattern: "^Foo"
+            }]
+        }],
+        errors: [{
+            type: "ExportNamedDeclaration",
+            line: 1,
+            column: 10,
+            endColumn: 13,
+            message: "'Bar' import from 'foo' is restricted because only imports that match the pattern '/^Foo/u' are allowed."
+        }]
+    },
```

Comment:
Can you also add a test like this but with a custom message, to test that code path?